python/ray/autoscaler/_private/kuberay/node_provider.py

```diff
@@ -263,16 +275,28 @@ def create_node(
         """Creates a number of nodes within the namespace."""
         url = "rayclusters/{}".format(self.cluster_name)
         raycluster = self._get(url)
-        group_name = tags["ray-user-node-type"]
+        group_name = tags[TAG_RAY_USER_NODE_TYPE]
         group_index = _worker_group_index(raycluster, group_name)
+        group_max_replicas = _worker_group_max_replicas(raycluster, group_index)
         tag_filters = {TAG_RAY_USER_NODE_TYPE: group_name}
         current_replica_count = len(self.non_terminated_nodes(tag_filters))
+        target_replica_count = current_replica_count + count
+        # Cap the replica count to maxReplicas.
+        if group_max_replicas is not None and group_max_replicas < target_replica_count:
+            # This is a race condition. We need a more permanent solution.
+            # See https://github.com/ray-project/kuberay/issues/560.
+            logger.warning(
+                "Autoscaler attempted to create "
+                "more than maxReplicas pods of type {}.".format(group_name)
+            )
+            target_replica_count = group_max_replicas
+
         path = f"/spec/workerGroupSpecs/{group_index}/replicas"
         payload = [
             {
                 "op": "replace",
                 "path": path,
-                "value": current_replica_count + count,
+                "value": target_replica_count,
```

Comment:
This line is great!

Reply:
Thanks! Purely mechanical change, since we had to extra processing for this value.